go/vt/tabletserver/query_executor.go

```diff
@@ -295,25 +295,33 @@ func (qre *QueryExecutor) execDDL() (*sqltypes.Result, error) {
 		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "DDL is not understood")
 	}
 
-	conn, err := qre.tsv.te.txPool.LocalBegin(qre.ctx)
+	if qre.transactionID != 0 {
+		conn, err := qre.tsv.te.txPool.Get(qre.transactionID, "for query")
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Recycle()
+		result, err := qre.execSQL(conn, qre.query, false)
+		if err != nil {
+			return nil, err
+		}
+		conn.ReloadSchemaOnCommit = true
+		return result, nil
+	}
+
+	result, err := qre.execAsTransaction(func(conn *TxConnection) (*sqltypes.Result, error) {
```

Comment:
I think it was because I was not doing a commit or rollback on error -- prior to this PR we were deferring the LocalCommit, so it always happened even on failure. I had changed it to an explicit LocalCommit on success. Something with the extra load on travis was causing execSql to fail and the commit never happened and the connection never returned to the pool.  Using execAsTransaction properly handles cleanup now. That's my guess at least.